### PR TITLE
Update all tool versions to support protocol 7

### PR DESCRIPTION
**Pull request by Victor-N-Suadicani**
## Purpose

Bumps all the tool versions to release the versions that support protocol 7.

## Changes

* Updated the Rust SDK to the 5.0.0 release.
* Got rid of various minor warnings.
* Bumped version of almost all the tools (I did not include the generator as that seems to have a protocol-7-supporting release already).

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by lassemand — APPROVED** [review]
LGTM

**Comment by lassemand** [comment]
Could you also bump the notification server version?

File: id-verifier/Cargo.lock [hunk 1]
```diff
@@ -604,7 +604,7 @@ checksum = "acbf1af155f9b9ef647e42cdc158db4b64a1b61f743629225fde6f3e0be2a7c7"
 
 [[package]]
```
Comment (DOBEN) [inline id-verifier/Cargo.lock:605]: bumping the version of the `id-verifier`?

**Comment by Victor-N-Suadicani** [comment]
Ah yes, missed it cause it didn't have a changelog file. I added one for it.

**Review by DOBEN — APPROVED** [review]

File: id-verifier/Cargo.lock (hunk 1 above)
Comment (Victor-N-Suadicani) [reply id-verifier/Cargo.lock:605]: Did not have a changelog either, I bumped it now and added the changelog.

**Pull request merged**